app/controllers/spree/orders_controller_decorator.rb

```diff
@@ -130,6 +131,24 @@ def cancel
 
   private
 
+  # Updates the various denormalized total attributes of the order and
+  # recalculates the shipment taxes
+  def update_totals_and_taxes
+    @order.updater.update_totals
+    @order.shipment.ensure_correct_adjustment_with_included_tax if @order.shipment
```

Comment:
this method will change in spree 2.2 https://github.com/spree/spree/commit/9548723ece02fd58c10e05d6a1bce69fe6b78ece#diff-a3fc88c87f17b31a30219f559940b2eaR258

Do you think order.update_shipping_fees! would work here? Or do you want to get rid of order.update_shipping_fees! ?

Reply:
The actual fix is to execute https://github.com/openfoodfoundation/openfoodnetwork/blob/d8710d283ae4649d7207423f16bc1479a2861223/app/models/spree/shipment_decorator.rb#L10-L16 so that we update the `included_tax` attribute. That is what `shipment.save` was triggering, remember?

I'll create an issue for `#update_shipping_fees!` taking that spree 2.2 into consideration. Thanks for that commit!

Reply:
Done in https://github.com/openfoodfoundation/openfoodnetwork/issues/3375